File: hotspot_analysis/2026-06-05_35yo_optimization/social-cards/render.mjs
import { chromium } from 'playwright-core';
import { join } from 'path';

const HTML = join(process.cwd(), 'index.html');
const OUT = join(process.cwd(), 'output');

(async () => {
  const browser = await chromium.launch({
    executablePath: '/opt/microsoft/msedge/msedge',
    headless: true,
    args: ['--no-sandbox', '--disable-gpu', '--font-render-hinting=none'],
  });

  const page = await browser.newPage();
  await page.setViewportSize({ width: 2400, height: 15000 });
  await page.goto(`file://${HTML}`, { waitUntil: 'domcontentloaded', timeout: 60000 });
  await page.waitForTimeout(5000); // 等待 WebGL canvas 和字体渲染

  // 截图 21:9 封面
  const wide = page.locator('#wechat-21x9');
  await wide.screenshot({ path: join(OUT, 'wechat-21x9-cover.png'), type: 'png' });

  // 截图 1:1 封面
  const square = page.locator('#wechat-1x1');
  await square.screenshot({ path: join(OUT, 'wechat-1x1-cover.png'), type: 'png' });

  // 截图配对预览
  const preview = page.locator('.pair-preview');
  await preview.screenshot({ path: join(OUT, 'wechat-cover-pair-preview.png'), type: 'png' });

  await browser.close();
  console.log('渲染完成，输出保存在:', OUT);
})();

File: 案例/social-cards/2026-06-16_body_entropy/render.py
from playwright.sync_api import sync_playwright
import os, subprocess, sys

html_path = os.path.join(os.getcwd(), 'index.html')
out_dir = os.path.join(os.getcwd(), 'output')

with sync_playwright() as p:
    browser = p.chromium.launch(
        executable_path='/opt/microsoft/msedge/msedge',
        headless=True,
        args=['--no-sandbox', '--disable-gpu', '--font-render-hinting=none']
    )
    page = browser.new_page()
    page.set_viewport_size({'width': 2400, 'height': 15000})
    page.goto(f'file://{html_path}', wait_until='domcontentloaded', timeout=60000)
    page.wait_for_timeout(5000)
    page.locator('#wechat-21x9').screenshot(path=os.path.join(out_dir, 'wechat-21x9-cover.png'))
    page.locator('#wechat-1x1').screenshot(path=os.path.join(out_dir, 'wechat-1x1-cover.png'))
    browser.close()

subprocess.run([sys.executable, os.path.join(os.getcwd(), 'stitch.py')], check=True)
print('渲染完成，输出保存在:', out_dir)
